### PR TITLE
Add initial version of example generator

### PR DESCRIPTION
**Pull request by planger**
#### What it does

Adds a new extension that can generate example projects into the current workspace.
As an initial starting point, it adds a plain CMake project example with launch.json, the example project of [`clangd-contexts`](https://github.com/eclipse-cdt-cloud/clangd-contexts), and traces for TraceCompass.cloud.

The example generation is triggered via a command. In the arguments of this command you can optionally already pre-specify the example type.

This extension is extensible in the sense that you can add more examples by adding a new example folder and extending the command asking you which example to generate.

Also
* Fix getting started page in browser app
* Fix browser app and add default workspace

Contributed on behalf of STMicroelectronics.

#### How to test
  * Build and run the CDT.cloud blueprint app
  * Run the command `Generate CDT.cloud Blueprint Example`
  * It'll ask you which example to generate
  * It'll generate the example into the workspace
  * It'll open the `README.md`
  * Test whether the instructions of the shown readme work as expected

  * Show the welcome page
  * Find the link in the text to create an example project (it should open the command and let you select the example type)
  * Alternatively, there is also a section on the right in the welcome page that directly creates the CMake example

### CODE REVIEW

File: theia-extensions/blueprint-example-generator/resources/cmake-example/CMAKE_EXAMPLE_README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+## Features to try out
+
+Below are some features that can be tried out:
+
+- Syntax highlighting
+- Building the code (via `Binary Build` task)
+- Code completion (after building)
+- Cross-file navigation (after building)
+- Debugging (after building)
```
Comment (sdirix) [inline theia-extensions/blueprint-example-generator/resources/cmake-example/CMAKE_EXAMPLE_README.md:17]: I think we should also mention the memory inspector?

File: theia-extensions/blueprint-example-generator/resources/example-traces/EXAMPLE_TRACES_README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Workspace with example traces
+
+This workspace contains two example traces that can be opened using the [CDT.cloud theia trace-extension](https://github.com/eclipse-cdt-cloud/theia-trace-extension).
+
+## Open a trace with the trace view
+
+1. Make sure, that the trace server path is set correctly in the Trace Viewer settings (Settings -> Extensions -> Trace Viewer -> Trace Server: Path)
```
Comment (sdirix) [inline theia-extensions/blueprint-example-generator/resources/example-traces/EXAMPLE_TRACES_README.md:7]: We should also mention where the trace viewer can be downloaded as Step 0.

**Review by sdirix — COMMENTED** [review]
Works fine in general!

 - Shall the examples be extracted into the root workspace? I would have expected that each of them is contained in an own directory after generation. It's hard to see what is what after generating them all into root. (Maybe not force them into directories but each example should if possible go into one)

File: applications/browser/workspace/.theia/settings.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "trace Viewer.trace Server.path": ""
+}
```
[the remark this answers is not in the document]
Comment (sgraband) [reply applications/browser/workspace/.theia/settings.json:3]: I added it to check in the `workspace` folder, but i can replace it with a `.gitkeep` file.

File: theia-extensions/blueprint-example-generator/resources/cmake-example/include/Rectangle.h [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class Rectangle {
```
[the remark this answers is not in the document]
Comment (sgraband) [reply theia-extensions/blueprint-example-generator/resources/cmake-example/include/Rectangle.h:1]: @planger ?

File: theia-extensions/blueprint-example-generator/resources/example-traces/EXAMPLE_TRACES_README.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# Workspace with example traces
```
[the remark this answers is not in the document]
Comment (sgraband) [reply theia-extensions/blueprint-example-generator/resources/example-traces/EXAMPLE_TRACES_README.md:1]: @planger ?

File: theia-extensions/blueprint-example-generator/resources/clangd-contexts/build.sh [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+# http://www.eclipse.org/legal/epl-2.0.
+#
+# This Source Code may also be made available under the following Secondary
+# Licenses when the conditions for such availability set forth in the Eclipse
+# Public License v. 2.0 are satisfied: GNU General Public License, version 2
+# with the GNU Classpath Exception which is available at
+# https://www.gnu.org/software/classpath/license.html.
+#
+# SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+#################################################################################
```
[the remark this answers is not in the document]
Comment (sgraband) [reply theia-extensions/blueprint-example-generator/resources/clangd-contexts/build.sh:16]: @planger ?

[the remark this answers is not in the document]
Comment (planger) [reply theia-extensions/blueprint-example-generator/resources/clangd-contexts/build.sh:16]: Yes, for all `clangd-contexts` files we can add the `Copyright (c) 2021 STMicroelectronics and others.` license header.

File: theia-extensions/blueprint-example-generator/resources/example-traces/EXAMPLE_TRACES_README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (planger) [reply theia-extensions/blueprint-example-generator/resources/example-traces/EXAMPLE_TRACES_README.md:1]: Puh, these traces originally come from: https://github.com/tuxology/tracevizlab/tree/master/labs
Actually this doesn't specify any license.
Hm, let's remove this example for now and ask Matthew to readd it?

File: theia-extensions/theia-blueprint-product/src/browser/branding-util.tsx [hunk 7]
```diff
@@ -56,15 +58,24 @@ export function renderWhatIs(windowService: WindowService): React.ReactNode {
...
+                role={'button'}
+                tabIndex={0}
+                onClick={() => generateExample(commandService)}
+                onKeyDown={(e: React.KeyboardEvent) => generateExample(commandService)}>
+                {'example project'}
+            </a>.</div>
+        <div>Explore the features, such as code editing, building, build configurations, debugging, etc.</div>
         <div>Download and open <ExternalBrowserLink text="example traces"
             url="https://github.com/tuxology/tracevizlab/blob/master/labs/TraceCompassTutorialTraces.tgz" windowService={windowService} ></ExternalBrowserLink> with
             Trace Compass Cloud.</div>
```
[the remark this answers is not in the document]
Comment (sgraband) [reply theia-extensions/theia-blueprint-product/src/browser/branding-util.tsx:71]: Actually not needed atm, as i removed the traces example due to unkown license.

[the remark this answers is not in the document]
Comment (sdirix) [reply theia-extensions/theia-blueprint-product/src/browser/branding-util.tsx:71]: I would not remove the traces example? Isn't this the license: https://github.com/tuxology/tracevizlab/blob/master/LICENSE?

The two examples here are part of the https://github.com/tuxology/tracevizlab/blob/master/labs/TraceCompassTutorialTraces.tgz file

File: theia-extensions/blueprint-example-generator/resources/example-traces/EXAMPLE_TRACES_README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (planger) [reply theia-extensions/blueprint-example-generator/resources/example-traces/EXAMPLE_TRACES_README.md:1]: Ah, there is a license, thanks @sdirix for catching that.
https://github.com/tuxology/tracevizlab/blob/master/LICENSE
https://github.com/eclipse-cdt-cloud/cdt-cloud-blueprint/pull/24#discussion_r910661739

**Review by sdirix — COMMENTED** [review]
Works for me!

Minor: The browser app does no longer start with an empty workspace.
Also it seems that the Mac CI build is broken?

**Pull request merged**